raiden-ts/src/channels/reducer.ts

```diff
@@ -34,19 +34,24 @@ const tokens = createReducer(initialState.tokens).handle(
     state[token] === tokenNetwork ? state : { ...state, [token]: tokenNetwork },
 );
 
+function withoutAction(pendingTxs: readonly ConfirmableAction[], action: ConfirmableAction) {
+  return pendingTxs.filter(
+    (a) => a.type !== action.type || action.payload.txHash !== a.payload.txHash,
+  );
+}
+
 const pendingTxs: Reducer<RaidenState['pendingTxs'], RaidenAction> = (
   state = initialState.pendingTxs,
   action: RaidenAction,
 ): RaidenState['pendingTxs'] => {
   // filter out non-ConfirmableActions's
   if (!ConfirmableAction.is(action)) return state;
-  // if confirmed==undefined, add action to state
-  else if (action.payload.confirmed === undefined) return [...state, action];
+  // if confirmed==undefined, deduplicate and add action to state
+  else if (action.payload.confirmed === undefined)
+    return [...withoutAction(state, action), action];
```

Comment:
Nitpick: this function is **doing** something. So I would use a verb for it like `removeAction()`. The first time I read this (during the meeting) I didn't understood what this should mean. So I actually had to read the function itself to understand the idea. Now it makes sense. Anyways I think this could be improved. There is enough logic you need to understand. Little supporting hints like this are appreciated.

Reply:
@weilbith This reminds of one of the most important books about software programming I have ever read (Code Complete by Steve McConnell - Chapter: Good Routine Names: https://flylib.com/books/en/2.823.1.63/1/).

Reply:
Done